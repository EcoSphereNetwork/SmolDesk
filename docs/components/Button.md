---
title: Button
description: 
---

## Function
TBD

## Props

| Name | Type |
| --- | --- |
| `label` | `string` |
| `onClick` | `() => void` |

## Source

[src/components/Button.tsx](/src/components/Button.tsx)
