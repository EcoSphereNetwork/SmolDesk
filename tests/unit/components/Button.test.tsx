import React from 'react';
import { composeStory } from '@storybook/testing-react';
import * as stories from '../../../src/components/Button.stories';
import { render, screen } from '@testing-library/react';

declare module '@storybook/react' {
  interface StoryObj<TArgs = any> {
    args?: Partial<TArgs>;
  }
}

const Primary = composeStory(stories.Primary, stories.default);

test('renders button with label', () => {
  render(<Primary />);
  expect(screen.getByRole('button')).toHaveTextContent('Submit');
  expect(document.body.innerHTML).toMatchSnapshot();
});
