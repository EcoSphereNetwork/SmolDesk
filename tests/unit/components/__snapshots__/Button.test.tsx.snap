// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders button with label 1`] = `"<div><button>Submit</button></div>"`;
